File: src/Framework/Frame.fs
namespace Animations

type public Frame(width: int, height: int) =
    
    let bytesPerPixel = 4 //currently only BGRA is supported
    let bytes = Array.zeroCreate<byte>(width * height * bytesPerPixel)

    member this.Width = width
    member this.Height = height
    member this.BytesPerRow = width * bytesPerPixel
    member this.Bytes = bytes

File: Animations.Framework/Drawing/Frame.cs
namespace Animations.Framework.Drawing
{
    public class Frame
    {
        public Frame(int width, int height)
        {
            Width = width;
            Height = height;
            Bytes = new byte[BytesPerRow * height];
        }

        private const int BytesPerPixel = 4; //currently only BGRA is supported
        
        public int Width { get; }
        public int Height { get; }
        public int BytesPerRow => Width * BytesPerPixel;
        public byte[] Bytes { get; }
    }
}